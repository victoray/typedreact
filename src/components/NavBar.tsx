import React from "react";

const NavBar: React.FC = () => {
    return (
        <div className={"ui top attached menu"}>
            <div className="ui category search item">
                <div className="ui transparent icon input">
                    <input className="prompt" type="text" placeholder="Search..."/>
                    <i className="search link icon"/>
                </div>
            </div>
            <div className="right menu">
                <div className="ui simple dropdown  item">
                    <img className="ui avatar image" src={"https://semantic-ui.com/images/avatar/small/matt.jpg"}
                         alt={""}/>
                    <span>Username</span> <i className="dropdown icon"/>
                    <div className="menu">
                        <div className="item">Logout <i className="sign-out icon"/></div>
                    </div>
                </div>
            </div>
        </div>);
};

export default NavBar;
