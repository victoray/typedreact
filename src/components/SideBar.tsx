import React from "react";
import {Link} from "react-router-dom";

const SideBar: React.FC = () => {
    return (
        <div className="ui left fixed vertical menu">
            <div className="item">
                <img className="ui mini image" alt={""} src={"https://semantic-ui.com/images/logo.png"}/>
            </div>
            <Link className="item" to={"/"}>DashBoard <i className="chart line icon"/></Link>
            <Link className="item" to={"/"}>StarWars <i className="star icon"/></Link>
            <Link className="item" to={"/"}>Avengers <i className="nintendo switch icon"/></Link>
            <Link className="item" to={"/"}>Settings <i className="cogs icon"/></Link>
        </div>);
};

export default SideBar;
