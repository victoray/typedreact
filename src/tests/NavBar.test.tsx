import React from "react";
import renderer from "react-test-renderer";
import SideBar from "../components/SideBar";
import {BrowserRouter} from "react-router-dom";
import NavBar from "../components/NavBar";


it("should render correctly", () => {
    const tree = renderer
        .create(<NavBar visible={false}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("should render correctly with sidebar", () => {
    const sidebar = renderer
        .create(<BrowserRouter><SideBar visible={true}/><NavBar visible={true}/></BrowserRouter>)
        .toJSON();
    expect(sidebar).toMatchSnapshot();
});
