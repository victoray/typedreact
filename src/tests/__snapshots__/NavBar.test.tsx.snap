// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="ui top fixed menu borderless"
>
  <div
    className="link item"
  >
    <img
      alt=""
      className="ui mini image"
      src="https://semantic-ui.com/images/logo.png"
    />
  </div>
  <div
    className="link item"
    onClick={[Function]}
  >
    <i
      className="sidebar icon"
    />
    Menu
  </div>
  <div
    className="ui  category search item"
  >
    <div
      className="ui large icon input"
    >
      <input
        className="prompt"
        placeholder="Search..."
        type="text"
      />
      <i
        className="search link icon"
      />
    </div>
  </div>
  <div
    className="right menu"
  >
    <div
      className="ui simple dropdown  item"
    >
      <img
        alt=""
        className="ui avatar image"
        src="https://semantic-ui.com/images/avatar/small/matt.jpg"
      />
      <span>
        Username
      </span>
       
      <i
        className="dropdown icon"
      />
      <div
        className="menu"
      >
        <div
          className="item link"
        >
          Logout 
          <i
            className="sign-out icon"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render correctly with sidebar 1`] = `
Array [
  <div
    className="ui visible sidebar left fixed vertical menu"
  >
    <div
      className="item"
    >
      <img
        alt=""
        className="ui mini image"
        src="https://semantic-ui.com/images/logo.png"
      />
    </div>
    <a
      className="item"
      href="/dashboard"
      onClick={[Function]}
    >
      DashBoard 
      <i
        className="chart line icon"
      />
    </a>
    <a
      className="item"
      href="/starwars"
      onClick={[Function]}
    >
      StarWars 
      <i
        className="star icon"
      />
    </a>
    <a
      className="item"
      href="/avengers"
      onClick={[Function]}
    >
      Avengers 
      <i
        className="nintendo switch icon"
      />
    </a>
    <a
      className="item"
      href="/settings"
      onClick={[Function]}
    >
      Settings 
      <i
        className="cogs icon"
      />
    </a>
  </div>,
  <div
    className="ui top fixed menu borderless"
  >
    <div
      className="link item"
    >
      <img
        alt=""
        className="ui mini image"
        src="https://semantic-ui.com/images/logo.png"
      />
    </div>
    <div
      className="link item"
      onClick={[Function]}
    >
      <i
        className="sidebar icon"
      />
      Menu
    </div>
    <div
      className="ui  category search item"
    >
      <div
        className="ui large icon input"
      >
        <input
          className="prompt"
          placeholder="Search..."
          type="text"
        />
        <i
          className="search link icon"
        />
      </div>
    </div>
    <div
      className="right menu"
    >
      <div
        className="ui simple dropdown  item"
      >
        <img
          alt=""
          className="ui avatar image"
          src="https://semantic-ui.com/images/avatar/small/matt.jpg"
        />
        <span>
          Username
        </span>
         
        <i
          className="dropdown icon"
        />
        <div
          className="menu"
        >
          <div
            className="item link"
          >
            Logout 
            <i
              className="sign-out icon"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
